pkg/routeagent_driver/handlers/healthchecker/healthchecker.go

```diff
@@ -0,0 +1,325 @@
+		}
+
+		logger.Infof("HealthChecker is already running for %q - stopping", endpoint.Name)
+		pingerObject.Stop()
+		delete(h.pingers, endpoint.Spec.CableName)
+	}
+
+	pingerConfig := pinger.Config{
+		IP:                 endpoint.Spec.HealthCheckIP,
+		MaxPacketLossCount: h.config.MaxPacketLossCount,
+	}
+
+	if h.config.PingInterval != 0 {
+		pingerConfig.Interval = time.Second * time.Duration(h.config.PingInterval)
+	}
+
+	if h.config.MaxPacketLossCount != 0 {
+		pingerConfig.MaxPacketLossCount = h.config.MaxPacketLossCount
+	}
+
+	newPingerFunc := h.config.NewPinger
+	if newPingerFunc == nil {
+		newPingerFunc = pinger.NewPinger
+	}
+
+	pingerObject := newPingerFunc(pingerConfig)
+	h.pingers[endpoint.Spec.CableName] = pingerObject
+	pingerObject.Start()
+
+	logger.Infof("HealthChecker started pinger for CableName: %q with HealthCheckIP %q",
+		endpoint.Spec.CableName, endpoint.Spec.HealthCheckIP)
+
+	return nil
+}
+
+func (h *controller) RemoteEndpointDeleted(endpoint *submarinerv1.Endpoint) error {
+	h.Lock()
+	defer h.Unlock()
+
+	if pingerObject, found := h.pingers[endpoint.Spec.CableName]; found {
+		pingerObject.Stop()
+		delete(h.pingers, endpoint.Spec.CableName)
+	}
+
+	return nil
+}
+
+func (h *controller) Init() error {
+	go func() {
+		wait.Until(func() {
+			h.Lock()
+			defer h.Unlock()
+
+			h.syncRouteAgentStatus()
+		}, RouteAgentUpdateInterval, h.stopCh)
+	}()
+
+	return nil
+}
+
+// TransitionToNonGateway is called once for each transition of the local node from Gateway to a non-Gateway.
+func (h *controller) TransitionToNonGateway() error {
+	h.Lock()
+	defer h.Unlock()
+
+	if h.config.HealthCheckerEnabled {
+		remoteEndpoints := h.State().GetRemoteEndpoints()
+
+		for i := range remoteEndpoints {
+			err := h.processEndpointCreatedOrUpdated(&remoteEndpoints[i])
+			if err != nil {
+				logger.Warningf("Error processing remote endpoint %s: %v", remoteEndpoints[i].Name, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// TransitionToGateway is called once for each transition of the local node from non-Gateway to a Gateway.
+func (h *controller) TransitionToGateway() error {
+	h.Lock()
+	defer h.Unlock()
+
+	if h.config.HealthCheckerEnabled {
+		for i := range h.pingers {
+			h.pingers[i].Stop()
+			delete(h.pingers, i)
+		}
+	}
+
+	h.syncRouteAgentStatus()
+
+	return nil
+}
+
+func (h *controller) GetNetworkPlugins() []string {
+	return []string{event.AnyNetworkPlugin}
+}
+
+func (h *controller) GetName() string {
+	return "routeAgent-health-checker"
+}
+
+func (h *controller) syncRouteAgentStatus() {
+	routeAgent := h.generateRouteAgentObject()
+	routeAgent.Status.RemoteEndpoints = []submarinerv1.RemoteEndpoint{}
+	remoteEndpoints := h.State().GetRemoteEndpoints()
+
+	for i := range remoteEndpoints {
+		pingerObject, found := h.pingers[remoteEndpoints[i].Spec.CableName]
+		if !found {
+			logger.Warningf("Pinger not found for %q", remoteEndpoints[i].Spec.CableName)
+			continue
+		}
+
+		latencyInfo := pingerObject.GetLatencyInfo()
```

Comment:
There won't be a pinger if `!h.config.HealthCheckerEnabled` or `h.State().IsOnGateway()` so this needs to be after L256.